desktop/src/main/java/bisq/desktop/util/validation/PhoneNumberValidator.java

```diff
@@ -0,0 +1,143 @@
+package bisq.desktop.util.validation;
+
+import bisq.core.locale.Res;
+import bisq.core.util.validation.InputValidator;
+
+import lombok.Getter;
+
+import javax.annotation.Nullable;
+
+/**
+ * Performs lenient validation of international phone numbers, and transforms given
+ * input numbers into E.164 international form.  The E.164 normalized phone number
+ * can be accessed via {@link #getNormalizedPhoneNumber} after successful
+ * validation -- {@link #getNormalizedPhoneNumber} will return null if validation
+ * fails.
+ * <p>
+ * Area codes and mobile provider codes are not validated, but all numbers following
+ * calling codes are included in the normalized number.
+ * <p>
+ * @see  bisq.desktop.util.validation.CountryCallingCodes
+ */
+public class PhoneNumberValidator extends InputValidator {
+    /**
+     * ISO 3166-1 alpha-2 country code
+     */
+    private final String isoCountryCode;
+    /**
+     * Immutable map of international calling codes.
+     */
+    private final CountryCallingCodes countryCallingCodes;
+    /**
+     * The international calling code mapped to the 'isoCountryCode' constructor argument.
+     */
+    @Nullable
+    @Getter
+    private final String callingCode;
+    /**
+     * The normalized (digits only) representation of an international calling code.
+     */
+    private final String normalizedCallingCode;
+    /**
+     * Phone number in E.164 format.
+     */
+    @Nullable
+    @Getter
+    private String normalizedPhoneNumber;
+
+    // Hide no-arg constructor
+    private PhoneNumberValidator() {
+        this.isoCountryCode = null;
```

Comment:
OK.  I thought all final instance variables had to be initialized in a constructor, and in this class, the no-arg constructor had to be hidden (private) to prevent another class from using it.  In the former case, maybe it was true in an earlier JDK?  In the latter, I'm still not sure it was a good idea to remove the private no-arg constructor, but I followed your advice anyway.

Reply:
Sorry! I didn't mean to imply removing private constructor. I do agree it is important here.

So please either revert back to assigning nulls **or** simply get rid off the problematic final. In the context of clean code (small methods and classes) Java finals are overrated. They only protect the reference anyway and not the object's content.

Whatever you pick it will be OK with me.